src/Microsoft.ML.TorchSharp/NasBert/TextClassificationTrainer.cs

```diff
@@ -328,13 +328,17 @@ public Trainer(TextClassificationTrainer parent, IChannel ch)
                 var vocabulary = Tokenizer.Vocabulary;
                 vocabulary.AddMaskSymbol();
 
+                // Get row count and figure out num of unique labels
+                var rowCount = GetRowCountAndSetLabelCount(input);
+
                 // Initialize the model and load pre-trained weights
                 Model = new TextClassificationModel(_parent._options, vocabulary, _parent._options.NumberOfClasses);
                 Model.GetEncoder().load(GetModelPath());
                 Model.train();
 
                 // Figure out if we are running on GPU or CPU
                 Device = ((IHostEnvironmentInternal)_parent._host).GpuDeviceId != null && cuda.is_available() ? CUDA : CPU;
+                Contracts.Assert(Device == CPU || (((IHostEnvironmentInternal)_parent._host).FallbackToCpu != false || Device != CPU), "Fallback to CPU is false but no GPU detected");
```

Comment:
>Contracts.Assert [](http://example.com/codeflow?start=16&length=16)

Why using contracts.Assert and not Debug.Assert?

Reply:
We use our `Contracts` class in all our code. https://github.com/michaelgsharp/machinelearning/blob/4715e61879fc3c41fde7ca8b9fc4b304ee37d4c6/src/Microsoft.ML.Core/Utilities/Contracts.cs. Not sure what the real difference between that and `Debug.Assert` is.

Reply:
In .NET core, we are not using contracts any more.